src/runtime_src/core/tools/xbmgmt2/OO_UpdateShell.cpp

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_UpdateShell.h"
+
+// XRT - Include Files
+#include "core/common/query_requests.h"
+#include "core/common/system.h"
+#include "flash/flasher.h"
+#include "tools/common/XBUtilitiesCore.h"
+#include "tools/common/XBUtilities.h"
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/filesystem.hpp>
+#include <boost/format.hpp>
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <fstream>
+#include <iostream>
+// =============================================================================
+
+// ----- H E L P E R M E T H O D S ------------------------------------------
+
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+OO_UpdateShell::OO_UpdateShell(const std::string &_longName, const std::string &_shortName, bool _isHidden )
+    : OptionOptions(_longName,
+                    _shortName,
+                    "Update the shell partition for a 2RP platform",
+                    boost::program_options::value<decltype(plp)>(&plp)->implicit_value("all")->required(),
+                    "The partition to be loaded.  Valid values:\n"
+                      "  Name (and path) of the partition.",
+                    _isHidden)
+{
+  m_optionsDescription.add_options()
+    ("device,d", po::value<decltype(m_device)>(&m_device), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+    ("help", po::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+}
+
+static void
+program_plp(const xrt_core::device* dev, const std::string& partition)
+{
+  std::ifstream stream(partition.c_str(), std::ios_base::binary);
```

Comment:
```suggestion
  std::ifstream stream(partition, std::ios_base::binary);
```
Use a global substitution pass...